### ISSUE TITLE
raise numpy error

### ISSUE THREAD

**Issue by zoeith** [numba/numba#1636]
In numba 0.22.1,
```python
raise np.linalg.LinAlgError
```
worked fine, but with 0.23 I get the following:
```
NotImplementedError: raise $61.3

Failed at nopython (nopython frontend)
raise $61.3
```
Full traceback: https://gist.github.com/zoeith/a9a4376a431dae50d4ea
I got a slightly different error while trying to put together an mwe...
```
NotImplementedError: raise $12.3

Failed at nopython (nopython frontend)
raise $12.3
```
Here's the original function and mwe attempt: https://gist.github.com/zoeith/0c1438486bfd031b39d0

**Comment by pitrou** [comment]
This can be easily worked around using `from numpy.linalg import LinAlgError` and then (inside your function) `raise LinAlgError`.

I'd rather not fix this before the refactor in PR #1622 is merged, as that requires touching the same internal mechanism (constant inference).

**Comment by seibert** [comment]
OK, we'll address this bug in the Numba 0.24 release in February, but not put this into the bug fix 0.23.1 release happening this week.
